.github/workflows/pipeline.yml

```
@@ -0,0 +1,23 @@
+name: Deployment pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize]
+    #check when PR is open
+
+jobs:
+  simple_deployment_pipeline:
```

maybe rename this to more descriptive? 
```suggestion
   react_tests:
```